netty/src/main/java/io/grpc/transport/netty/NettyServerStream.java

```diff
@@ -70,12 +70,22 @@ void inboundDataReceived(ByteBuf frame, boolean endOfStream) {
 
   @Override
   public void request(final int numMessages) {
-    channel.eventLoop().execute(new Runnable() {
-      @Override
-      public void run() {
-        requestMessagesFromDeframer(numMessages);
-      }
-    });
+    if (channel.eventLoop().inEventLoop()) {
+      // Processing data read in the event loop so can call into the deframer immediately and since
```

Comment:
So much duplication :/ ... we should really revisit our class/interface hierarchy at some point.